app/models/reports/lsa/fy2019/base.rb

```diff
@@ -70,39 +72,238 @@ def value_for_options options
       display_string
     end
 
-    protected
-
-    def project_id_string options
+    private def project_id_string options
       str = ''
       if options['project_id'].present?
         if options['project_id'].is_a?(Array)
           if options['project_id'].delete_if(&:blank?).any?
-            str = "; Projects: #{options['project_id'].map{|m| GrdaWarehouse::Hud::Project.find_by_id(m.to_i)&.name || m if m.present?}.compact.join(', ')}"
+            str = "; Projects: #{options['project_id'].map do |m|
+              GrdaWarehouse::Hud::Project.find_by_id(m.to_i)&.name || m if m.present? # rubocop:disable Metrics/BlockNesting
+            end.compact.join(', ')}"
           end
         else
-          str = "; Project: #{GrdaWarehouse::Hud::Project.find_by_id(options['project_id'].to_i)&.name || options['project_id'] }"
+          str = "; Project: #{GrdaWarehouse::Hud::Project.find_by_id(options['project_id'].to_i)&.name || options['project_id']}"
         end
       end
-      return str
+      str
     end
 
-    def project_group_string options
+    private def project_group_string options
       if (pg_ids = options['project_group_ids']&.compact) && pg_ids&.any?
         names = GrdaWarehouse::ProjectGroup.where(id: pg_ids).pluck(:name)
-        if names.any?
-          return "; Project Groups: #{names.join(', ')}"
-        end
+        return "; Project Groups: #{names.join(', ')}" if names.any?
       end
       ''
     end
 
-    def sub_population_string options
+    private def sub_population_string options
       if (sub_population = options['sub_population']) && sub_population.present?
         return "; Sub Population: #{sub_population.humanize.titleize}"
       end
       ''
     end
 
+    def missing_data(user)
+      @missing_data = {}
+      @range = ::Filters::DateRange.new(start: Date.current - 3.years, end: Date.current)
```

Comment:
Maybe report range and back 3 years?

Reply:
This gets run before the report is queued, so there's no date range available.  Checking for enrollments in the past 3 years should be sufficient.